### ISSUE TITLE
Visualization the loss

### ISSUE THREAD

**Issue by twmht** [loshchil/AdamW-and-SGDW#4]
Hi,

how do you visualize the loss? what kind of tools you used to plot?

![_072](https://user-images.githubusercontent.com/1567200/46657235-673e8c80-cbe2-11e8-92fc-1459f036e343.png)

It seems that it definitely helps a a lot if we want to tune the learning rate and the weight decay.

thanks!

**Comment by loshchil** [comment]
You need to collect Z values (here, validation error) for different (X,Y) values (here, weight decay and learning rate) to then use (X,Y,Z) to make a **contour or surf plot** (e.g., here in MATLAB).

**Comment by twmht** [comment]
@loshchil 

is this similar to https://github.com/tomgoldstein/loss-landscape?

**Comment by loshchil** [comment]
I guess that they also make 3D contour/surf plots based on XYZ data.
